src/components/Context.js

```diff
@@ -1,10 +1,41 @@
 import React, { Component } from 'react';
 import { storeProducts, detailProduct } from '../data';
-import client from './Cart/VoucherifyContext';
 import { toast } from 'react-toastify';
 
 const ProductContext = React.createContext();
 
+const addPromotion = 'ADD_PROMOTION';
+
+const reducer = (action) => (state, props) => {
+  switch (action.type) {
+    case addPromotion:
+      console.log(state.cartDiscount);
+
+      if (state.cartDiscount !== []) {
+        if (state.cartDiscount[1] === 'percent') {
```

Comment:
inside [0] is a number,

so e.g [10, "percent"] or [10, "amount']

I could reverse it so it would be more obvious that first we check if the value is a percent or amount and then we calculate how big is a discount.